internal/websocket/ws.go

```diff
@@ -359,37 +372,38 @@ func (ws *Websocket) dial(ctx context.Context, mode ipMode) (*nhws.Conn, *http.R
 }
 
 type custRT struct {
-	transport http.Transport
+	transport *http.Transport
 }
 
 func (rt *custRT) RoundTrip(r *http.Request) (*http.Response, error) {
 	return rt.transport.RoundTrip(r)
 }
 
-// updateClientTransport updates the http client's Transport and set the DialContext's
-// named network as the provided `mode`.
-func (ws *Websocket) updateClientTransport(mode ipMode) {
-	// Override client's Transport with custRT (reusing certain configurations)
-	// and update it's DialContext with the provided mode.
-	transport := ws.client.Transport.(*http.Transport)
-	ws.client.Transport = &custRT{
-		transport: http.Transport{
-			Proxy:                 http.ProxyFromEnvironment,
-			MaxIdleConns:          1,
-			MaxIdleConnsPerHost:   1,
-			MaxConnsPerHost:       1,
-			TLSHandshakeTimeout:   transport.TLSHandshakeTimeout,
-			ExpectContinueTimeout: transport.ExpectContinueTimeout,
-		},
+// newHTTPClient returns a HTTP client using the provided `mode` as its named network.
+func (ws *Websocket) newHTTPClient(mode ipMode) (*http.Client, error) {
+	client, err := ws.httpClientConfig.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
+	// Override NewClient()'s Transport and update it's DialContext with the provided mode.
+	transport, err := ws.httpClientConfig.Transport.NewTransport(&arrangetls.Config{})
+	if err != nil {
+		return nil, err
 	}
+
+	transport.Proxy = http.ProxyFromEnvironment
```

Comment:
arrangehttp.ClientConfig.Transport lacks the http.Transport.Proxy so we set it here